src/js/services/conversation-service.js

```diff
@@ -16,12 +16,14 @@ export function sendMessage(text) {
             text: text,
             role: 'appUser'
         };
+        const state = store.getState();
+        if (state.conversation.messages.length === 0 &&
```

Comment:
this doesn't work because you can have appMaker messages and whispers before an appUser message.

But I almost fixed it the same way you did ;) I created an `handleFirstUserMessage` function and appended it to the promise chain after calling core.sendmessage.